addons/udes_stock_packaging/models/container_type.py

```diff
@@ -64,9 +64,16 @@ def _check_dimensions_and_weight(self):
         """
         Ensure these fields do not contain negative quantities else
         it could impact calculations when the container types are used.
+
+        Exception is made for no box (use_product_dims = True)
         """
+
         non_negative_fields = ["length", "width", "height", "weight"]
-        violating_records = self.filtered(lambda ct: any([getattr(ct, f) <= 0 for f in non_negative_fields]))
+        violating_records = self.filtered(
+            lambda ct: not getattr(ct, "use_product_dims", False) and any(
```

Comment:
Why is needed getattr, the field use_product_dims is added in this module so i think not needed to use getattr
```suggestion
            lambda ct: not ct.use_product_dims and any(
```